src/main/java/noppes/npcs/scripted/interfaces/IWorld.java

```diff
@@ -199,7 +218,20 @@ public interface IWorld {
      * @return Returns the entity which was spawned
      */
     public ScriptEntity spawnClone(int x, int y, int z, int tab, String name);
-
+    
+    public boolean spawnEntity(Entity e);
+    
+    /**
+	 * Create a new default faction
+	 * @param name The name of the new faction. If the name already exists an '_' will be added at the end
+	 * @return The faction object which was created
+	 */
+    public ScriptFaction createFaction(String name);
+    
+    public ScriptFaction getFactionFromName(String name);
+    
+    public ScriptFaction getFactionFromId(int id);
+    
```

Comment:
Faction functions in ScriptWorld aren't needed anymore, handled by API.getFactions(), which returns a function controller object.